modules/mesos/src/main/java/org/apache/ignite/mesos/resource/IgniteProvider.java

```diff
@@ -52,40 +75,99 @@ public IgniteProvider(String downloadFolder) {
 
     /**
      * @param ver Ignite version.
-     * @throws IOException If downloading failed.
      * @return Path to latest ignite.
+     * @throws IOException If downloading failed.
      */
     public String getIgnite(String ver) throws IOException {
         return downloadIgnite(ver);
     }
 
     /**
      * @param ver Ignite version which will be downloaded. If {@code null} will download the latest ignite version.
-     * @throws IOException If downloading failed.
      * @return Ignite archive.
+     * @throws IOException If downloading failed.
      */
     public String downloadIgnite(String ver) throws IOException {
         assert ver != null;
 
         URL url;
 
+        // get the latest version.
         if (ver.equals(ClusterProperties.DEFAULT_IGNITE_VERSION)) {
-            URL updateUrl = new URL(DOWNLOAD_LINK);
+            try {
+                ver = findLatestVersion();
 
-            HttpURLConnection conn = (HttpURLConnection)updateUrl.openConnection();
+                // and try to retrieve from a mirror.
+                url = new URL(String.format(findMirror() + IGNITE_PATH, ver, ver));
+            }
+            catch (Exception e) {
+                // fallback to archive.
+                url = new URL(String.format(DOWNLOAD_URL_PATTERN, ver, ver));
```

Comment:
If findLatestVersion() fails then we will try to download https://archive.apache.org/dist/ignite/latest/apache-ignite-fabric-latest-bin.zip which is 404. We should somehow push the latest known version from pom.xml to source in order to fall back to it.

Reply:
@alamar I doubt fetching from pom.xml is very clean and improve much.

We can do
```
          catch (Exception e) {
                if (ver.equalsIgnoreCase(ClusterProperties.DEFAULT_IGNITE_VERSION))
                    throw new RuntimeException("Failed to identify the latest version.");
                
                // fallback to archive.
                url = new URL(String.format(DOWNLOAD_URL_PATTERN, ver, ver));
            }
```
Just let's fail (it's not critical) if nothing else can be done -- the user will notice and specify the version explicitly to download from the archive or from a custom URL.